src/network/networkd-link.c

```diff
@@ -651,6 +651,12 @@ static Link *link_free(Link *link) {
 
         hashmap_free(link->bond_slaves);
 
+        if (link->network) {
+                if (link->network->bond &&
+                    link_get(link->manager, link->network->bond->ifindex, &master) >= 0)
```

Comment:
these two if expressions could be merged into one?

Reply:
Yeah, but a later commit adds similar expression.